.github/workflows/mac_python_matrix_test.yml

```diff
@@ -12,9 +12,9 @@ jobs:
     runs-on: macos-latest
     strategy:
       fail-fast: false
-      max-parallel: 5
+      max-parallel: 4
```

Comment:
What is the implication of leaving max-parallel at 5 or even slightly higher.   
From what I see, currently we have to change the max-parallel every time we add or remove a python version.

Reply:
I don't think there is any implication in this case. I was simply maintaining the "Can't run more or fewer jobs than Python versions" precedent that we (nonverbally) have created. I'm happy to put it back to 5.

Reply:
I can see the argument for both cases... @blnicho do you have an opinion?

Reply:
Out of curiosity, what happens if you leave `max-parallel` out entirely?  Does it default to 1 or to unlimited?

Reply:
I just tested it - it looks like it is "unlimited" (in that I think you still can't run more than 20 workflows simultaneously, so if there are some prior ones pending, it'll wait for a job to finish).

Reply:
I would then just leave out out... @jsiirola do you agree?

Reply:
Yes.

Reply:
I would also vote to leave it out based on @mrmundt's comment that the default is "unlimited"

Reply:
Has been corrected in the newest commit